Update artifact generation page (#48)
gatling/frontline-cloud-doc: content/reference/user/artifact_gen/index.md

---
title: "Artifact Generation"
description: "Learn how to generate an artifact for Gatling Enterprise from the Gatling zip bundle, or from a Maven, SBT or Gradle project."
lead: "Generate an artifact from your Gatling bundle or with a Maven, SBT or Gradle project."
date: 2021-03-08T13:50:32+01:00
lastmod: 2021-03-08T13:50:32+01:00
weight: 10040
---

## Generating Artifacts for FrontLine

Gatling Enterprise deploys packages containing your compiled Simulations and resources.
Those packages have to be generated upstream.
Gatling Enterprise is compatible with Gatling 3.3, 3.4, 3.5 and 3.6.

### Gatling zip bundle

> Copy the [`artifact.sh`](https://raw.githubusercontent.com/gatling/gatling/master/gatling-bundle/src/universal/bin/artifact.sh) or [`artifact.bat`](https://raw.githubusercontent.com/gatling/gatling/master/gatling-bundle/src/universal/bin/artifact.bat) file in the `bin` directory of your unzipped Gatling bundle.

Reviewer: Already present in gatling bundle 3.6.x
We can remove that

Author: Done; I added a warning that 3.6.0+ is required in this section.

---
title: "Artifact Generation"

> description: "Learn how to generate an artifact for Gatling Enterprise from the Gatling zip bundle, or from a Maven, SBT or Gradle project."

Reviewer 2: ```suggestion
description: "Learn how to generate an artifact for Gatling Enterprise from the Gatling zip bundle, or from a Maven, SBT, or Gradle project."
```

Author: Oh, the good old Oxford comma debate. 😅 Let's go for it, it would be useful in more complex sentences and it's usually not recommended to switch styles.

> lead: "Generate an artifact from your Gatling bundle or with a Maven, SBT or Gradle project."

Reviewer 2: ```suggestion
lead: "Generate an artifact from your Gatling bundle or with a Maven, SBT, or Gradle project."
```

date: 2021-03-08T13:50:32+01:00
lastmod: 2021-03-08T13:50:32+01:00
weight: 10040
---

## Generating Artifacts for FrontLine

Gatling Enterprise deploys packages containing your compiled Simulations and resources.
Those packages have to be generated upstream.
Gatling Enterprise is compatible with Gatling 3.3, 3.4, 3.5 and 3.6.

### Gatling zip bundle

Run the script `artifact.sh` (on Linux or macOS) or `artifact.bat` (on Windows), found in the `bin` directory of your unzipped Gatling bundle.
This will generate a `target/artifact.jar` file. You can then upload this file in the [Artifacts section]({{< ref "../artifact_conf" >}}).

{{< alert warning >}}

> Requires the Gatling bundle version 3.6.0 or superior.

Reviewer: Hum.
Hard to tell that it requires 3.6.0 when we wrote (L14) that it's compatible from 3.3...

Author: This is specifically in the "gatling bundle" section, and it says "gatling bundle" in the sentence. Line 14 is in the general introduction, it includes the other methods of generating a package. Maybe it can be clarified a little bit more though.

The main alternative I see is to add back the old explanation for older versions of Gatling bundle? Something like this:
```
Run the script `artifact.sh` (on Linux or macOS) or `artifact.bat` (on Windows), found in the `bin` directory of your unzipped Gatling bundle.
This will generate a `target/artifact.jar` file. You can then upload this file in the [Artifacts section]({{< ref "../artifact_conf" >}}).

{{< alert warning >}}
The scripts are included in the Gatling bundle version 3.6.0 or superior. For older versions, you will need to copy the
[`artifact.sh`](https://raw.githubusercontent.com/gatling/gatling/master/gatling-bundle/src/universal/bin/artifact.sh)
or [`artifact.bat`](https://raw.githubusercontent.com/gatling/gatling/master/gatling-bundle/src/universal/bin/artifact.bat)
files to the `bin` directory of your unzipped Gatling bundle.
{{< /alert >}}
```

Reviewer: Sadly, yep

### SBT Project

In your SBT project configuration, you must add:

- the Gatling dependencies

> - the Gatling Enterprise SBT plugin (`"io.gatling.frontline" % "sbt-frontline"`), used to package your code into a deployable artifact

Reviewer: ```suggestion
- the Gatling Enterprise SBT plugin (`"io.gatling.frontline" % "sbt-frontline"`), packages your code into a deployable artifact
```

Author: Not grammatically correct in this context. But it could be "[the plugin], **which** packages your code..."; that could work. 🤔

Reviewer: I prefer ", which packages"

val feeder = csv("foo.csv").shard
```

Assuming the CSV file contains 1000 entries, and you run your simulation on 3 Gatling nodes, the entries will be distributed as follows:

- First node will access the first 333 entries
- Second node will access the next 333 entries
- Third node will access the last 334 entries

{{< alert tip >}}
`shard` is available in Gatling OSS DSL but is a noop there. It's only effective when running tests with Gatling Enterprise.
{{< /alert >}}

## Resolving Injector Location in Simulation

When running a distributed test from multiple locations, you could be interested in knowing where a given injector is deployed in order to trigger specific behaviors depending on the location.

> For example, you might want to hit `https://example.co.uk` if the injector is deployed in the `Europe - Paris` AWS region, and `https://example.com` otherwise.

Reviewer 2: ```suggestion
For example, you might want to hit `https://example.co.fr` if the injector is deployed in the `Europe - Paris` AWS region, and `https://example.com` otherwise.
```

In your simulation code, you can resolve the name of the pool the current injector is deployed in:

```scala
val poolName = System.getProperty("gatling.frontline.poolName")

> val baseUrl = if (poolName == "Europe - Paris") "https://example.co.uk" else "https://example.com"

Reviewer 2: ```suggestion
val baseUrl = if (poolName == "Europe - Paris") "https://example.co.fr" else "https://example.com"
```

Author: `example.fr`, `example.co.fr` doesn't exist. 😉
